server/src/main/java/io/crate/expression/BaseImplementationSymbolVisitor.java

```diff
@@ -53,14 +54,12 @@ public BaseImplementationSymbolVisitor(TransactionContext txnCtx, Functions func
 
     @Override
     public Input<?> visitFunction(Function function, C context) {
-        FunctionIdent ident = function.info().ident();
         Signature signature = function.signature();
-        FunctionImplementation functionImplementation;
-        if (signature == null) {
-            functionImplementation = functions.getQualified(ident);
-        } else {
-            functionImplementation = functions.getQualified(signature, ident.argumentTypes());
-        }
+        List<DataType> argumentTypes = Symbols.typeView(function.arguments());
```

Comment:
I think `argumentTypes` is only used in the error case further below. Maybe the `typeView` call could be moved into the `else` block.